src/mono/mono/metadata/metadata.c

```diff
@@ -1470,18 +1486,18 @@ mono_metadata_decode_row_col_raw (const MonoTableInfo *t, int idx, guint col)
 {
 	const char *data;
 	int n;
+	guint8 column_offset;
 
 	guint32 bitfield = t->size_bitfield;
 
 	g_assert (GINT_TO_UINT32(idx) < table_info_get_rows (t));
 	g_assert (col < mono_metadata_table_count (bitfield));
 	data = t->base + idx * t->row_size;
 
-	n = mono_metadata_table_size (bitfield, 0);
-	for (guint i = 0; i < col; ++i) {
-		data += n;
-		n = mono_metadata_table_size (bitfield, i + 1);
-	}
+	n = mono_metadata_table_size (bitfield, col);
+	column_offset = t->column_offsets [col];
+	g_assert (column_offset < 0xFFu);
```

Comment:
Does this assert anything different than what line 1494 asserts: `g_assert (col < mono_metadata_table_count (bitfield));` ?

Reply:
it's effectively the same assertion, it's just validating the contents of the lookup table in addition to the input parameter